app/forms/mappings/Mappings.scala

```diff
@@ -30,8 +30,9 @@ trait Mappings extends Formatters with Constraints {
   protected def localDate(invalidKey: String,
                           dayKey: String,
                           monthKey: String,
-                          yearKey: String): FieldMapping[LocalDate] =
-    of(new LocalDateFormatter(invalidKey, dayKey, monthKey, yearKey, Seq.empty))
+                          yearKey: String,
+                          validDateKey: String): FieldMapping[LocalDate] =
```

Comment:
Better to rename it to invalidDateKey as this key will be used for invalid date.